src/listener/Link.php

```diff
@@ -21,7 +21,7 @@ public function process(Line $line)
     {
         $link = $line->getAttribute('link');
         if ($link) {
-            $this->updateInput($line, '<a href="'.$link.'" target="_blank">'.$line->input.'</a>');
+            $this->updateInput($line, '<a href="'.$line->getLexer()->escape($link).'" target="_blank">'.$line->escapedInput().'</a>');
```

Comment:
I was thinking of changing this to:

```suggestion
            $this->updateInput($line, '<a href="'.$line->escapedAttribute('link').'" target="_blank">'.$line->escapedInput().'</a>');
```

And use a similar `escapedInsertJsonKey($key)` for block listeners.

Reply:
So i think this `$link = $line->getAttribute('link');` would then be `$link = $line->getEscapedAttribute('link');` to be consistent? Sounds good to me!

Reply:
Good point.

I am thinking that for a next major we should make this more explicit to not make mistakes in accidentally not escaping:

```php
Line->getAttribute() // escapes by default
Line->getUnsafeAttribute() // no escaping
Line->getInput() // escapes by default
Line->getUnsafeInput() // no escaping
Line->getInsertJsonKey() // escapes by default
Line->getUnsafeInsertJsonKey() // no escaping
```

Then in the next major we could remove the `escaped` methods and the `escapeInput` option. If we want that, we could prepare this by adding the `unsafe` methods already.

Reply:
Perfect, yes i agree on this.

Reply:
We could do it now i assume, as its anyhow processed trough `escapeInput` in lexer method, so as we turn this off by default it will have the same response but also means that in this version 

+ Line->getAttribute() // escapes by default (unless escapeInput is false)
+ Line->getUnsafeAttribute() // no escaping

have the same response unless you are not turning on escapeInput in lexer object

Reply:
:+1:

Reply:
I figured that this only works for `getInput()` & `getUnsafeInput()`. The `attribute` and `insertJsonKey` variants don't necessarily return a string value, and will be way harder to escape in a proper way.

So I think we'll have to leave it like this. Until we use another tool, twig alike, to mix html & non-html which handles this correctly.